rmgpy/reduction/reduction.py

```diff
@@ -268,13 +268,18 @@ def searchTargetIndex(targetLabel, reactionModel):
     """
     Searches for the Species object in the core species
     of the reaction that has the same label as the parameter string.
-    """
+    reactionModel must be of class CoreEdgeReactionModel
 
+    Has known issues dealing with duplicate labels. See reductionTest.py
+    for a unittest of this issue.
+    """
+    spc_labels = []
     for i, spc in enumerate(reactionModel.core.species):
         if spc.label == targetLabel:
             return i
+        spc_labels.append(spc.label)
 
-    raise Exception('{} could not be found...'.format(targetLabel))
+    raise Exception('{} could not be found in species labels {}.'.format(targetLabel,spc_labels))
```

Comment:
I don't think it's a good way to print out all the species labels in the core; some mechanism can be pretty big and printing out is not so helpful.

If we are concerned about the situation where two species have same labels, maybe we should use isomorphism check instead. An idea would be to require user to feed SMILES instead of labels in the reduction_input.py